mito-ai/mito_ai/gemini_client.py

```diff
@@ -0,0 +1,70 @@
+# Copyright (c) Saga Inc.
+# Distributed under the terms of the GNU Affero General Public License v3.0 License.
+
+from typing import AsyncGenerator
+from google import genai
+from mito_ai.completions.models import AgentResponse
+
+
+class GeminiClient:
+    def __init__(self, api_key: str, model: str):
+        self.client = genai.Client(api_key=api_key)
+        self.model = model
+
+    async def generate_content(self, contents: str) -> str:
```

Comment:
Maybe we can call this function `request_completion` to keep it in sync with the language in the provider.py file